packages/algob/src/builtin-tasks/compile.ts

```diff
@@ -1,13 +1,112 @@
+import type { Algodv2, CompileOut } from "algosdk";
+import { readdirSync, readFileSync, writeFileSync } from 'fs';
+import * as path from 'path';
+import YAML from "yaml";
+
 import { task } from "../internal/core/config/config-env";
-// import { createClient } from "../lib/driver";
-// import { AlgobRuntimeEnv, TaskArguments } from "../types";
+import { parseAlgorandError } from "../internal/core/errors";
+import { assertDir, ASSETS_DIR, CACHE_DIR } from "../internal/core/project-structure";
+import { createClient } from "../lib/driver";
+import type { AlgobRuntimeEnv, ASCCache, Network } from "../types";
 import { TASK_COMPILE } from "./task-names";
+const murmurhash = require('murmurhash'); // eslint-disable-line @typescript-eslint/no-var-requires
+
+const tealExt = ".teal";
 
 export default function (): void {
-  task(TASK_COMPILE, "Compilation task")
-    .setAction(
-      async () => {
-        console.log("compiling......");
-      }
-    );
+  task(TASK_COMPILE, "Compile all TEAL smart contracts")
+    .addFlag("force", "recompile even if the source file didn't change")
+    .setAction(_compile);
+}
+
+export interface TaskArgs {
+  force: boolean
+}
+
+async function _compile ({ force }: TaskArgs, env: AlgobRuntimeEnv): Promise<void> {
+  const op = new CompileOp(env.network);
+  return await compile(force, op);
+}
+
+export async function compile (force: boolean, op: CompileOp): Promise<void> {
+  await assertDir(CACHE_DIR);
+  const cache = readArtifacts(CACHE_DIR);
+
+  for (const f of readdirSync(ASSETS_DIR)) {
+    if (!f.endsWith(tealExt)) { continue; }
+
+    let c = cache.get(f);
+    const [teal, thash] = readTealAndHash(path.join(ASSETS_DIR, f));
+    if (!force && c !== undefined && c.srcHash === thash) {
+      console.log(`smart-contract source "${f}" didn't change, skipping.`);
```

Comment:
Let's unify how we use logs. 
Either change the old ones or these ones. We already have two differents methods for it and one of them (the old one) doesn't output anything.
I think we should have a task for that.

Reply:
What's the other way to print info messages to the user? `debug` is used for debugging.
What I see, in other places (eg the `init` command and `project-creation.ts`) `console.log` is used for info logs.  If you want to change it then you can create a task in the icebox.

Reply:
I didn't use debug logging and this statement looks like it's too verbose for normal operation where user simply reruns the compilation.
So I started thinking whether we should do something about it.
I don't know if I want a task about it.